Build breaks on old versions of GCC

I tried building on an old Linux machine running gcc 4.6.3 (2011) and GNU make 3.81 (2006) and ran into failures:

```sh
cc -c -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o md5.o md5.c
cc -c -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o sha256.o sha256.c
cc -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o nsfinfo nsfinfo.c md5.o sha256.o
nsfinfo.c: In function 'my_MD5_File':
nsfinfo.c:107:2: error: 'for' loop initial declarations are only allowed in C99 mode
nsfinfo.c:107:2: note: use option -std=c99 or -std=gnu99 to compile your code
nsfinfo.c: In function 'my_SHA256_File':
nsfinfo.c:163:2: error: 'for' loop initial declarations are only allowed in C99 mode
make: *** [nsfinfo] Error 1
```

I added `-std=c99` to the `CFLAGS` and then I got POSIX related failures instead:
```sh
nsfinfo.c: In function 'main':
nsfinfo.c:235:2: error: implicit declaration of function 'getopt' [-Werror=implicit-function-declaration]
nsfinfo.c:235:2: error: nested extern declaration of 'getopt' [-Werror=nested-externs]
nsfinfo.c:247:10: error: 'optind' undeclared (first use in this function)
nsfinfo.c:247:10: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors
make: *** [nsfinfo] Error 1
```

--- a/nsfinfo.c
+++ b/nsfinfo.c
@@ -58,6 +58,7 @@ my_MD5_File(const char *filename)
 {
 	FILE *fd;
 	size_t filelen;
+	unsigned int i;
 	unsigned char *filebuf = NULL;
 	MD5_CTX c;
 	BYTE hash[MD5_BLOCK_SIZE];
@@ -102,7 +103,7 @@ my_MD5_File(const char *filename)
 	hashstr = calloc((MD5_BLOCK_SIZE * 2) + 1, 1);
 	hashstrp = &hashstr[0];
 
-	for (unsigned int i = 0; i < MD5_BLOCK_SIZE; i++) {
+	for (i = 0; i < MD5_BLOCK_SIZE; i++) {
 		hashstrp += sprintf(hashstrp, "%02x", hash[i]);
 	}
 
@@ -114,6 +115,7 @@ my_SHA256_File(const char *filename)
 {
 	FILE *fd;
 	size_t filelen;
+	unsigned int i;
 	unsigned char *filebuf = NULL;
 	SHA256_CTX c;
 	BYTE hash[SHA256_BLOCK_SIZE];
@@ -158,7 +160,7 @@ my_SHA256_File(const char *filename)
 	hashstr = calloc((SHA256_BLOCK_SIZE * 2) + 1, 1);
 	hashstrp = &hashstr[0];
 
-	for (unsigned int i = 0; i < SHA256_BLOCK_SIZE; i++) {
+	for (i = 0; i < SHA256_BLOCK_SIZE; i++) {
 		hashstrp += sprintf(hashstrp, "%02x", hash[i]);
 	}
 

